### PR TITLE
Add "salt" beacon

### LINKED ISSUE #56461: Feature Request: New beacon "monitor"
This beacon would provide a similar functionality as job scheduling but would implement it within salt's beacon system. The user would supply a list of _salt_ execution modules to run with optional args and kwargs. The execution modules return would be included with the event and the minion event would only be fired if the return is "truthy" (i.e. not `None`, `False`, `0`, or similar)

### PR DESCRIPTION
**Pull request by jtraub91**
### What does this PR do?
Adds new beacon: `salt_monitor`
### What issues does this PR fix or reference?
Closes https://github.com/saltstack/salt/issues/56461
### New Behavior
Allows user to specify one or more salt function (optionally with args / kwargs) to execute every beacon interval. The beacon will fire an event only if the return from its function is "truthy" and the event will include the function call (with args /  kwargs, if applicable), as well as the return data.
### Tests written?
Yes
### Commits signed with GPG?
No

### CODE REVIEW

**Comment by jtraub91** [comment]
Did anyone get a chance to look at this PR?

**Comment by jtraub91** [comment]
@twangboy I fixed the failing tests. Can this PR be reviewed, please ?

**Comment by jtraub91** [comment]
@twangboy bump

File: salt/beacons/salt_monitor.py
Comment (jtraub91) [reply salt/beacons/salt_monitor.py:42]: `config` is an iterable due to a somewhat unfortunate convention used in the config files: the configuration is supplied as a list of single key dictionaries. So through the use of these couple of lines we convert that to a simple multi-key dictionary. The `list()` function is used because map is a generator in recent versions of python. The lines of code I used here can be found in other beacon modules, for example [inotify](https://github.com/saltstack/salt/blob/master/salt/beacons/inotify.py#L112-L113).

Comment (s0undt3ch) [reply salt/beacons/salt_monitor.py:42]: Thanks for the clarification.

File: salt/beacons/salt_monitor.py
Comment (jtraub91) [reply salt/beacons/salt_monitor.py]: Cool. Committed.

File: salt/beacons/salt_monitor.py
Comment (jtraub91) [reply salt/beacons/salt_monitor.py]: Good call. Committed.

**Review by xeacott — DISMISSED** [review]

**Comment by jtraub91** [comment]
@xeacott Does your approval mean this will go into Aluminum? Sorry to bug ya, but I'm just curious and wanted to make sure this gets included before it becomes too out of date with master.

**Comment by xeacott** [comment]
@jtraub91 Unfortunately no; we have a traditional ruling in the Core team that _most_ PR's require 3 approvals to be considered "done done" by us. I will kindly ping @sagetherage as she can provide a much better answer about this getting Aluminum.

**Comment by sagetherage** [comment]
Since @s0undt3ch requested changed I will ask for his re-review and I have updated this with master so it is running tests again, then likely it can be merged, these labels will also help :)

**Comment by sagetherage** [comment]
how many reviewers? well, if tests are passing and tests written, docs, and changelog then only 1 review is needed, but we do like review from those that have requested changes, and 3 approvals is mandatory for anything that doesn't have all of those - clearer? Maybe, I have labeled so Megan can review for merge.

**Review by garethgreenaway — DISMISSED** [review]

**Review by twangboy — DISMISSED** [review]

**Review by garethgreenaway — APPROVED** [review]

**Comment by max-arnold** [comment]
Apparently, the indentation after the `salt_fun` keyword should be increased:

```yaml
beacons:
  salt_monitor:
    - salt_fun:
      - slsutil.renderer:
          args:
            - salt://states/apache.sls
          kwargs:
            - default_renderer: jinja
      - test.ping
    - interval: 3600  # seconds
```

**Comment by jtraub91** [comment]
@max-arnold yes good catch that was a typo in the docstring! 😳

**Pull request merged**
